geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionAdvisor.java

```diff
@@ -453,10 +466,21 @@ public boolean initializationGate() {
     if (initialized) {
       return false;
     }
-    synchronized (initializeLock) {
-      if (!initialized) {
-        exchangeProfiles();
-        return true;
+    boolean exchangedProfiles = false;
+    try {
+      synchronized (initializeLock) {
+        if (!initialized) {
```

Comment:
Could this sync be replaced with a compareAndSet on initialized?

Reply:
I don't think that would be safe. If two thread come into this at the same time, then we want the second thread to wait for the first thread to finish the initialization. And we only want one thread to do the profile exchange. I don't see a performance problem with this existing code. Initialization happens once early and after that this is just a volatile read of a boolean which I think would perform the same as a compareAndSet call. Let me know if I'm missing something.